eis_qgis_plugin/processing/algorithms/filtering/frost_filter.py

```diff
@@ -0,0 +1,71 @@
+from qgis.core import (
+    QgsProcessingParameterNumber,
+    QgsProcessingParameterRasterDestination,
+    QgsProcessingParameterRasterLayer,
+)
+
+from eis_qgis_plugin.processing.eis_processing_algorithm import EISProcessingAlgorithm
+
+
+class EISFrostFilter(EISProcessingAlgorithm):
+    def __init__(self) -> None:
+        super().__init__()
+
+        self._name = "frost_filter"
+        self._display_name = "Frost filter"
+        self._group = "Filtering"
+        self._group_id = "filtering"
+        self._short_help_string = '''
+            Apply a Frost filter to the input raster.
+            Higher number of looks result in better edge preservation.
+            '''
+
+    def initAlgorithm(self, config=None):
+        self.alg_parameters = [
+            "input_raster",
+            "size",
+            "damping_factor",
+            "output_raster",
+        ]
+
+        input_raster_param = QgsProcessingParameterRasterLayer(
+                name=self.alg_parameters[0], description="Input raster"
+            )
+        input_raster_param.setHelp("The input raster data set.")
+        self.addParameter(input_raster_param)
+
+        window_size_param = QgsProcessingParameterNumber(
+            name=self.alg_parameters[1],
+            description="Size",
+            minValue=1,
+            defaultValue=3,
+        )
+        window_size_param.setHelp(
+            '''
+            The size of the filter window.
+            E.g., 3 means a 3x3 window.
+            '''
+        )
+        self.addParameter(window_size_param)
+
+        damping_factor = QgsProcessingParameterNumber(
+            name=self.alg_parameters[2],
+            description="Damping factor",
+            minValue=0.1,
+            defaultValue=1.0,
+            type=QgsProcessingParameterNumber.Double,
+        )
```

Comment:
Damping factor needs to be only > 0 according to EIS Toolkit docs, so I would put a smaller min value here, perhaps 0.001 would be enough